clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -2296,6 +2392,19 @@ protected DelegateConsumerRecordResult delegateConsumerRecord(
             controlMessage,
             consumerRecord.getPubSubMessageHeaders(),
             partitionConsumptionState);
+        /**
+         * For the local consumption, the batch data won't be produce to the local VT again, so we will switch
+         * to real-time writer upon EOP here and for the remote consumption of VT, the switch will be handled
+         * in the following section as it needs to flush the messages and then switch.
```

Comment:
Do we need to call flush in the below?

Reply:
In theory, we don't need to flush if the leader doesn't `produceToLocalKafka` and I am moving this section into the following `if` scope.